### PR TITLE
Add submodule docstrings

### LINKED ISSUE #113: Add documentation to importing module
- add short code example to all importing functions (currently only ``load_from_msp`` and ``load_from_usi`` have it)
- Add introduction text to readthedocs for importing module. Also mention there that other packages can be used to adapt parsing option (pymzml and pyteomics).

### LINKED ISSUE #126: Add submodule docstrings
We just added a docstring to the ``filtering\__init__.py`` which makes the submodule page nicer to read: https://matchms.readthedocs.io/en/latest/api/matchms.filtering.html

We should also do the same for ``importing\__init__.py``, ``exporting\__init__.py``, ``similarity\__init__.py``.
(Partly overlaps with issue #113)

### PR DESCRIPTION
**Pull request by florian-huber**
- add submodule docstrings to respective ``__init__.py``s
- add code example to exporting and importing functions (where missing).

See build: https://matchms.readthedocs.io/en/extend_submodule_docs/api/matchms.importing.html

### CODE REVIEW

**Comment by florian-huber** [comment]
Not sure why figure is not showing in the readthedocs for the ``filtering`` submodule. It actually did show in some earlier versions...
https://matchms.readthedocs.io/en/extend_submodule_docs/api/matchms.filtering.html

File: matchms/filtering/__init__.py [hunk 1]
```diff
@@ -26,6 +29,12 @@
 
     Maximum intensity is 1.00
 
+.. image:: ../_static/filtering_sketch.png
```
Comment (sverhoeven) [inline matchms/filtering/__init__.py]: Caption only supported by figure not by image directive.
```suggestion
.. figure:: ../_static/filtering_sketch.png
```

Comment (florian-huber) [reply matchms/filtering/__init__.py]: Oh, of course. Thanks!

File: matchms/exporting/save_as_json.py [hunk 2]
```diff
@@ -9,6 +9,23 @@ def save_as_json(spectrums: List[Spectrum], filename: str):
 
     :py:attr:`~matchms.Spectrum.losses` of spectrum will not be saved.
 
+    Example:
+
+    .. code-block:: python
+
+        from matchms import Spectrum
```
Comment (sverhoeven) [inline matchms/exporting/save_as_json.py:17]: I am getting numpy not found error when I run this.
```suggestion
        import numpy
        from matchms import Spectrum
```

File: matchms/exporting/save_as_mgf.py [hunk 3]
```diff
@@ -8,6 +8,23 @@ def save_as_mgf(spectrums: List[Spectrum], filename: str):
 
     :py:attr:`~matchms.Spectrum.losses` of spectrum will not be saved.
 
+    Example:
+
+    .. code-block:: python
+
+        from matchms import Spectrum
```
Comment (sverhoeven) [inline matchms/exporting/save_as_mgf.py:16]: Also missing numpy here
```suggestion
        import numpy
        from matchms import Spectrum
```

File: matchms/similarity/__init__.py [hunk 4]
```diff
@@ -1,4 +1,17 @@
-"""similarity module"""
+"""
+Functions for computing spectra similarities
+############################################
+
+Matchms provides a number of frequently used similarity scores to compare mass
+spectra. This includes scores based on comparing peak positions and intensities
```
Comment (sverhoeven) [inline matchms/similarity/__init__.py]: To make easier to read can this be reformatted into a bullet list?

**Comment by florian-huber** [comment]
Thanks Stefan, with the ``.. figure::`` command it not renders the filtering sketch. 
For now I simply took the figure from our paper. In the future I hope I can add some suitable illustrations for other modules as well (e.g. similarity), but not in this PR.

File: CHANGELOG.md [hunk 5]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Changed
 
 - Extended, updated, and corrected documentation for filter functions [#118](https://github.com/matchms/matchms/pull/118)
+- Add basic submodule documentation and more code examples [#128](https://github.com/matchms/matchms/pull/128)
```
Comment (sverhoeven) [inline CHANGELOG.md]: Remove `add` or move to `### Added` section.

**Review by sverhoeven — CHANGES_REQUESTED** [review]
Nice usage of links to functions and classes.

In importing examples please dont mix usage `list(), `next()` or no postprocessing. Use same example format each time.

File: CHANGELOG.md (hunk 5 above)
Comment (florian-huber) [reply CHANGELOG.md]: Moved it to ``### Added``.

File: matchms/similarity/__init__.py (hunk 4 above)
Comment (florian-huber) [reply matchms/similarity/__init__.py]: Done (need to check if it renders fine).

**Comment by florian-huber** [comment]
I changed the ``next(...)`` to a ``list(...)`` in the code examples. Still they will look slightly different, because not all importer functions are generators.

File: matchms/similarity/__init__.py (hunk 4 above)
Comment (sverhoeven) [reply matchms/similarity/__init__.py]: Renders ok

**Review by sverhoeven — APPROVED** [review]

**Comment by florian-huber** [comment]
Thanks for reviewing Stefan!

**Pull request merged**